test-utils/src/rollover.rs

```diff
@@ -0,0 +1,144 @@
+    assert_eq!(old_merkle_tree.canopy_depth, new_merkle_tree.canopy_depth);
+}
+
+pub fn assert_rolledover_merkle_trees_metadata(
+    old_merkle_tree_metadata: &MerkleTreeMetadata,
+    new_merkle_tree_metadata: &MerkleTreeMetadata,
+    current_slot: u64,
+    new_queue_pubkey: &Pubkey,
+) {
+    // Old Merkle tree
+    // 1. rolled over slot is set to current slot
+    // 2. next Merkle tree is set to the new Merkle tree
+
+    // New Merkle tree
+    // 1. index is equal to the old Merkle tree index
+    // 2. rollover fee is equal to the old Merkle tree rollover fee (the fee is calculated onchain in case rent should change the fee might be different)
+    // 3. tip is equal to the old Merkle tree tip
+    // 4. rollover threshold is equal to the old Merkle tree rollover threshold
+    // 5. rolled over slot is set to u64::MAX (not rolled over)
+    // 6. close threshold is equal to the old Merkle tree close threshold
+    // 7. associated queue is equal to the new queue
+    // 7. next merkle tree is set to Pubkey::default() (not set)
+    // 8. owner is equal to the old Merkle tree owner
+    // 9. delegate is equal to the old Merkle tree delegate
+
+    assert_eq!(
+        old_merkle_tree_metadata.access_metadata,
+        new_merkle_tree_metadata.access_metadata
+    );
+
+    assert_eq!(
+        old_merkle_tree_metadata.rollover_metadata.index,
+        new_merkle_tree_metadata.rollover_metadata.index
+    );
+    assert_eq!(
+        old_merkle_tree_metadata.rollover_metadata.rollover_fee,
+        new_merkle_tree_metadata.rollover_metadata.rollover_fee,
+    );
+    assert_eq!(
+        old_merkle_tree_metadata
+            .rollover_metadata
+            .rollover_threshold,
+        new_merkle_tree_metadata
+            .rollover_metadata
+            .rollover_threshold,
+    );
+    assert_eq!(
+        old_merkle_tree_metadata.rollover_metadata.network_fee,
+        new_merkle_tree_metadata.rollover_metadata.network_fee,
+    );
+    assert_eq!(
+        old_merkle_tree_metadata.rollover_metadata.rolledover_slot,
+        current_slot,
+    );
+    assert_eq!(
+        old_merkle_tree_metadata.rollover_metadata.close_threshold,
+        new_merkle_tree_metadata.rollover_metadata.close_threshold
+    );
+
+    assert_eq!(new_merkle_tree_metadata.associated_queue, *new_queue_pubkey);
+    assert_eq!(new_merkle_tree_metadata.next_merkle_tree, Pubkey::default());
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn assert_rolledover_queues_metadata(
+    old_queue_metadata: &QueueMetadata,
+    new_queue_metadata: &QueueMetadata,
+    current_slot: u64,
+    new_merkle_tree_pubkey: &Pubkey,
+    new_queue_pubkey: &Pubkey,
+    old_merkle_tree_lamports: u64,
+    new_merkle_tree_lamports: u64,
+    new_queue_lamports: u64,
+) {
+    assert_eq!(
+        old_queue_metadata.rollover_metadata.rolledover_slot,
+        current_slot
+    );
+    // Isn't this wrong???
```

Comment:
At some point we planned the index to be an identifier for a tree. If used as an identifier it should be carried over by rollovers in my opinion.